src/table/table-column-resizable.directive.ts

```diff
@@ -0,0 +1,239 @@
+                resizeMarkLine.destroy();
+                resizeOverlay.destroy();
+
+                this.renderWidthStyles(
+                  this.getWidthInRange(
+                    resizeRange,
+                    columnWidth + mouseUpEvent.clientX - initialMouseX,
+                  ),
+                );
+              }),
+            ),
+          );
+        }),
+      )
+      .subscribe(exec => {
+        exec();
+      });
+  }
+
+  private createResizeHandle() {
+    const resizeHandle: HTMLDivElement = this.renderer2.createElement('div');
+    this.renderer2.addClass(resizeHandle, resizableBem.element('handle'));
+    this.renderer2.appendChild(this.columnElement, resizeHandle);
+
+    return resizeHandle;
+  }
+
+  private createResizeMarkLine(initialOffset: number) {
+    const markLine: HTMLElement = this.renderer2.createElement('div');
+    this.renderer2.addClass(markLine, resizableBem.element('mark-line'));
+    this.renderer2.setStyle(
+      markLine,
+      'left',
+      initialOffset - markLineWidth + 'px',
+    );
+    if (this.isStickyLeftBorderColumn()) {
+      this.renderer2.addClass(markLine, 'inStickyBorderElemLeft');
+    }
+    this.renderer2.appendChild(this.containerElement, markLine);
+    return {
+      element: markLine,
+      updateOffset: (offset: number) => {
+        this.renderer2.setStyle(
+          markLine,
+          'left',
+          offset - markLineWidth + 'px',
+        );
+      },
+      destroy: () => {
+        this.renderer2.removeChild(this.containerElement, markLine);
+      },
+    };
+  }
+
+  private createResizeOverlay() {
+    const resizeOverlay = this.renderer2.createElement('div');
+    this.renderer2.addClass(resizeOverlay, resizableBem.element('overlay'));
+    this.renderer2.appendChild(this.containerElement, resizeOverlay);
+    return {
+      element: resizeOverlay,
+      destroy: () => {
+        this.renderer2.removeChild(this.containerElement, resizeOverlay);
+      },
+    };
+  }
+
+  private getColumnWidth() {
+    return this.columnElement.clientWidth;
+  }
+
+  private getColumnOffset() {
+    return (
+      this.columnElement.getBoundingClientRect().left -
+      this.containerElement.getBoundingClientRect().left
+    );
+  }
+
+  private getWidthInRange(
+    [minWidth, maxWidth]: [number, number],
+    width: number,
+  ): number {
+    return Math.min(Math.max(width, minWidth), maxWidth);
+  }
+
+  private getResizeRange(): [number, number] {
+    const minWidth = this.getActualWidth(this.minWidth);
+    const maxWidth = this.getActualWidth(this.maxWidth);
+    return [minWidth, maxWidth];
+  }
+
+  private getActualWidth(width: number | string): number {
+    if (typeof width === 'number') {
+      return width;
+    }
+    if (width.endsWith('%')) {
+      return (
+        (this.containerElement.clientWidth * parseInt(width.slice(0, -1))) / 100
+      );
+    }
+    if (width.endsWith('px')) {
+      return parseInt(width.slice(0, -2));
```

Comment:
```suggestion
      return Number.parseFloat(width.slice(0, -2));
```

怎么感觉被恢复成最开始的代码了？所有 review 后的变更都没了。。。